File: src/modules/user/domain/user.entity.ts
import { BeltProgress } from "src/modules/belt/domain/belt-progress.entity";
import { TrainingSession } from "src/modules/training/domain/training-session.entity";
import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from "typeorm";

@Entity()
export class User {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  name: string;

  @Column({ unique: true })
  email: string;

  @Column()
  password: string;

  @OneToMany(() => BeltProgress, (beltProgress) => beltProgress.user)
  beltProgress: BeltProgress[];

  @OneToMany(() => TrainingSession, (trainingSession) => trainingSession.user)
  trainingSessions: TrainingSession[];

  @Column({ type: 'timestamp', default: () => 'CURRENT_TIMESTAMP' })
  createdAt: Date;
}

File: src/modules/user/domain/user_entity.go
package domain_user

import (
	domain_belt "bjj-tracker/src/modules/belt/domain"
	domain_training "bjj-tracker/src/modules/training/domain"
	"time"
)

type User struct {
	ID               string                            `gorm:"primarykey"`
	Name             string                            `json:"name" gorm:"not null"`
	Username         string                            `json:"username" gorm:"not null;unique"`
	Avatar           string                            `json:"avatar" gorm:"not null"`
	Email            string                            `json:"email" gorm:"not null;unique"`
	Password         string                            `json:"password" gorm:"not null"`
	BeltProgress     []domain_belt.BeltProgress        `gorm:"foreignKey:UserID"`
	TrainingSessions []domain_training.TrainingSession `gorm:"foreignKey:UserID"`
	CreatedAt        time.Time                         `json:"created_at" gorm:"default:CURRENT_TIMESTAMP"`
	UpdatedAt        time.Time                         `json:"updated_at" gorm:"default:CURRENT_TIMESTAMP"`
}
